### DISCUSSION TITLE
Linux on Surface Pro 11th Edition?

### DISCUSSION THREAD

**Discussion by pablopussell** [linux-surface/linux-surface#1452] [General]
I am jumping the gun here, as this device has not yet been released to the general public, but am wondering, now that the new devices will be using ARM architecture, specifically Snapdragon Elite/Plus chipsets, is this anticipated to make Linux easier to port to these systems or harder?
Thanks for any insight

**Comment by StollD** [comment]
Most likely harder. You will have to write device tree definitions that describe the hardware of every device to the kernel, since most of the relevant Qualcomm drivers can't read that data from ACPI like an x86 driver would.

**Reply by pablopussell** [reply]
Interesting. Considering Android runs on a limited version of Linux and Android phones run on ARM architecture, I would have thought that could possibly make porting things easier, but my knowledge here is limited so I defer to your comment.

**Reply by qzed** [reply]
See also https://github.com/linux-surface/linux-surface/issues/1476.

**Comment by ThatOneCalculator** [comment]
https://github.com/kfechter/Linux-Surface-Copilot/

**Reply by HubKing** [reply]
How is it going? I'm interested in SL7, not SP11, but since they seem to use the same CPU, I wonder if you succeed it. I really don't want to use Windows 11.

**Comment by denisix** [comment]
I have SP11 with OLED, installed ArchLinux using this repo, https://github.com/dwhinham/linux-surface-pro-11
works well for daily routines, except no audio/touchscreen/pen/suspend

**Reply by ProgrammerIn-wonderland** [reply]
Just curious, do you have working touchpad?

**Reply by ooaklee** [reply]
If you don't mind using Ubuntu, you can try https://github.com/ooaklee/linux-surface-pro-11-oe#from-scratch-commands

I've got Wi-Fi, sound, Bluetooth working.

There's still a lot to sort out, but touchpad, suspend works and battery life is good.

**Reply by ooaklee** [reply]
@denisix Just an update, I have a kernel that gives full audio, mic, front webcam, touchscreen, pen, and more support

Check out my latest release: https://github.com/ooaklee/linux-surface-pro-11-oe/releases/tag/sp11-qcom-x1e-7.2.0-jg-0sp11v19. The only thing is you'll have to rebuild the add-on library for [iptsd sp11](https://github.com/ooaklee/linux-surface-pro-11-oe/releases/tag/sp11-iptsd-v1), since you're using Arch instead of a Debian-based distro. In fact, you'll have to compile a couple of things:
- https://github.com/ooaklee/linux-surface-pro-11-oe/tree/main/userspace/camera/libcamera
- https://github.com/ooaklee/linux-surface-pro-11-oe/tree/main/userspace/iptsd-sp11


See the [video here ](https://github.com/ooaklee/linux-surface-pro-11-oe/issues/16#issuecomment-5467184091)

**Comment by ooaklee** [comment]
https://github.com/ooaklee/linux-surface-pro-11-oe#from-scratch-commands

**Reply by adrianbartyczak** [reply]
@ooaklee I just installed Dale's Arch Linux image on my Surface Pro 11. However, the CPU runs hot all the time and the battery drains fast.

Does this happen with your Ubuntu image as well? Thanks

**Reply by ooaklee** [reply]
Hey @adrianbartyczak, the battery is pretty good and the CPUs runs OK. Only time it runs hot is when doing something intensive like compiling the kernel.

**Reply by adrianbartyczak** [reply]
Okay, I will try it out! Thanks!

**Reply by adrianbartyczak** [reply]
Hey @ooaklee, I tried setting up your Ubuntu image but I get an error when running build-sp11-qcom-x1e-kernel-docker.sh:

docker: Error response from daemon: failed to mount /tmp/containerd-mount784312865: mount source: "overlay", target: "/tmp/containerd-mount784312865", fstype: overlay, flags: 0, data: "workdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/8/work,upperdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/8/fs,lowerdir=/var/lib/containerd/io.containerd.snapshotter.v1.overlayfs/snapshots/1/fs,index=off", err: invalid argument

I think it's because I'm using Linux. Do you by any chance have workaround for Linux? Thanks

**Reply by ooaklee** [reply]
Hey @adrianbartyczak, just saw your message. Were you able to pass this stage? I've created this [how-to](https://github.com/ooaklee/linux-surface-pro-11-oe/blob/main/docs/how-to/how-to-troubleshoot-linux-docker-overlay.md), hope it helps

**Reply by adrianbartyczak** [reply]
That helped me fix the issue. Thanks. But I still get an error when running build-sp11-qcom-x1e-kernel-docker.sh:

```
...
fe924e3fb717: Pull complete
Digest: sha256:7cc5e35f6567ee8c66d2abb4aab0fd866669e6207c237c3a8f0947a5c7f17092
Status: Downloaded newer image for ubuntu:25.10
exec /work/docker-build-inside.sh: exec format error
Docker kernel build failed; inspect the log above for the first build error.
If the source tree was partially prepared, rerun with --reset-source after fixing the failure.
```

It looks like a variable is not getting set as it can't find the path before "/work".

I'm interested in skipping Docker entirely (as you mentioned in your how to document), but I don't know what `SP11DATA` should be set to.

**Reply by adrianbartyczak** [reply]
I set up a Mac VM and tried building the image. It went a lot better, as a lot of things were downloading and building, however, I got another Docker error:

```
...
Setting up g++-aarch64-linux-gnu (4:15.2.0-4ubuntu1) ...
Setting up libpython3-dev:arm64 (3.13.7-1) ...
Setting up python3.13-dev (3.13.7-1ubuntu0.4) ...
Setting up debhelper (13.24.2ubuntu1) ...
Setting up python3-dev (3.13.7-1) ...
Setting up equivs (2.3.2) ...
Setting up g++ (4:15.2.0-4ubuntu1) ...
update-alternatives: using /usr/bin/g++ to provide /usr/bin/c++ (c++) in auto mode
update-alternatives: warning: skip creation of /usr/share/man/man1/c++.1.gz because associated file /usr/share/man/man1/g++.1.gz (of link group c++) doesn't exist
Setting up build-essential (12.12ubuntu1) ...
Setting up liblwp-protocol-https-perl (6.14-1) ...
Setting up libwww-perl (6.78-1ubuntu0.1) ...
Setting up devscripts (2.25.19ubuntu2) ...
Processing triggers for libc-bin (2.42-0ubuntu3.1) ...
Cloning into '/linux-work/source/git-qcom-x1e-7.0'...
fatal: fetch-pack: invalid index-pack output
Docker kernel build failed; inspect the log above for the first build error.
If the source tree was partially prepared, rerun with --reset-source after fixing the failure.
```

**Reply by ooaklee** [reply]
@adrianbartyczak, check out these two `how-to's`; they should help you move past the reported issues you've run into:

- https://github.com/ooaklee/linux-surface-pro-11-oe/blob/main/docs/how-to/how-to-troubleshoot-docker-exec-format-error.md
- https://github.com/ooaklee/linux-surface-pro-11-oe/blob/main/docs/how-to/how-to-troubleshoot-kernel-git-clone-failures.md

If any improvements could be made to the docs, please create a PR so we can merge it for the next person who might get stuck

**Reply by adrianbartyczak** [reply]
I followed the steps and my docker is running correctly. Is this a real Surface Pro Ubuntu ARM build repository?

**Reply by ooaklee** [reply]
I'm using the Ubuntu Concept image from the community effort at https://discourse.ubuntu.com/t/ubuntu-concept-snapdragon-x-elite/48800 and extending it with Surface Pro 11 support — a custom device tree, patched qcom-x1e kernel packages, and firmware/support tooling. The result is a bootable Ubuntu ARM64 live USB and an installable system for the Surface
Pro 11 with Snapdragon X Elite.

Not sure what you mean by "real Surface Pro Ubuntu ARM build repository" — it's by no means an official Microsoft, Ubuntu, or linux-surface release. I'm using this repo to share what I've discovered and make it as easy as possible for others to learn from what worked (and what didn't).

**Reply by adrianbartyczak** [reply]
That does sound good. It looks like a lot of work. If you've got a working Ubuntu ARM64 image for the Surface Pro 11, can you send me the .img file?

**Reply by kyjus25** [reply]
Uploaded a copy of the .img in the thread over [here](https://github.com/dwhinham/linux-surface-pro-11/issues/28#issuecomment-4803378863)

TL;DR: Wi-Fi still broken for me on my X1P using this image, broke the Flex Keyboard functionality that otherwise worked on Ubuntu Concept (and stock Ubuntu 26.04) directly

**Comment by adrianbartyczak** [comment]
I created a set of instructions for installing Dale's Arch Linux kernel on the Surface Pro 11 [here](https://github.com/dwhinham/linux-surface-pro-11/issues/16#issuecomment-4764101723).

**Comment by adrianbartyczak** [comment]
[Full guide](https://github.com/adrianbartyczak/install-ubuntu-sp11) for installing Ubuntu on the Surface Pro 11.
